src/reconcile/rec_write.c

```diff
@@ -2620,25 +2626,28 @@ __rec_write_image(WT_SESSION_IMPL *session, WTI_RECONCILE *r, WTI_REC_CHUNK *chu
 
     page = r->page;
     multi = &r->multi[r->multi_next - 1];
-    block_meta = &page->block_meta;
+    block_meta = &page->disagg_info->block_meta;
```

Comment:
This flow is possible for ASC also, we do need the check for NULL here before accessing the block_meta.

Reply:
fixed